pyriemann/classification.py

```diff
@@ -511,6 +516,36 @@ def predict(self, covtest):
             the prediction for each trials according to the closest centroid.
         """
         dist = self._predict_distances(covtest)
-        neighbors_classes = self.classes_[np.argsort(dist)]
+        neighbors_classes = self.classmeans_[np.argsort(dist)]
         out, _ = stats.mode(neighbors_classes[:, 0:self.n_neighbors], axis=1)
         return out.ravel()
+
+    def predict_proba(self, X):
+        """Predict proba using softmax.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+
+        Returns
+        -------
+        prob : ndarray, shape (n_matrices, n_classes)
+            Probabilities for each class.
+        """
+        n_matrices, _, _ = X.shape
+
+        dist = self._predict_distances(X)
+        idx = np.argsort(dist)
+        dist_sorted = np.take_along_axis(dist, idx, axis=1)
```

Comment:
```suggestion
        dist_sorted = dist[:, idx[:, :self.n_neighbors]]
```
Should be enough I think

Reply:
I tried, but it does not work.